### ISSUE TITLE
Scrolling the list widget by page

### ISSUE THREAD

**Issue by francisdb** [ratatui/ratatui#1370]
## Problem

I would like to implement a pg-down and -up in my app but it seems that the actually rendered lines for a list widget is not available. So for now I just scroll by 10 items.

## Solution

Preferably build this into the state of the widget. Similar to `scroll_down_by`.

## Alternatives

Store/make available the currently shown number of lines 

## Additional context

There was some prior PR on for the table widget but was never finished #4 

Implementing wrapped scrolling with the current list widget is also a manual task. And the selection related method don't seem to check bounds.

**Comment by orhun** [comment]
I implemented a similar paging list for [binsider](https://github.com/orhun/binsider) before: https://github.com/orhun/binsider/commit/e86b10c72da39cc08ba0e6664fc4b12c859aadf6

But my reasoning was #1004

I guess it also makes sense to support this out of the box.

**Comment by francisdb** [comment]
This is what paging looks like using [dialoguer](https://github.com/console-rs/dialoguer) select.

* when you scroll past the last visible item the whole next page is shown and the selected item is on top
* when you scroll before the first visible item the whole previous page is shown and the selected item is on the bottom

Not saying this is the best solution but shows there is more to paging than just selecting the item x lines down.

https://github.com/user-attachments/assets/f7e74703-1388-455c-9d68-7e9460519f31

**Comment by orhun** [comment]
Yeah, I think supporting pagination like that would be really nice.

**Comment by joshka** [comment]
This is probably difficult to implement. I'd also like to see a bit of design for this before implementing if possible. Definitely some good ideas, but I'm not sure how feasible they would be to implement without breaking a lot of things.

**Comment by francisdb** [comment]
But the page size is depending on the render height, how would that info be propagated?

**Comment by orhun** [comment]
We can maybe call a function in the render loop (e.g. `state.set_page_height()`), wouldn't that work?

**Comment by joshka** [comment]
That would probably be ideal. We may also want to provide the item count to the state, which would enable properly moving to the end of the list (after first render at least, or after the method call).

**Comment by raylu** [comment]
noting that ratatui/ratatui#1553 (my fix for ratatui/ratatui#1514 which switches offset from number of list items to number of lines) makes scrolling by page very easy (just increase offset by height)
